python/sdk/merlin/resource_request.py

```diff
@@ -99,7 +101,10 @@ def gpu_name(self, gpu_name):
 
     def validate(self):
         if self._min_replica is None and self._max_replica is None:
-            return 
+            return
 
         if self._min_replica > self._max_replica:
```

Comment:
is it possible for user to not set min replica and set max replica?

Reply:
Yes, it's possible and it will use the default environment's default resource request value.

Reply:
for that case the below code will be failed right?
```
if self._min_replica > self._max_replica:
```
I assume set the default resource is set in server

Reply:
> I assume set the default resource is set in server

Not really. The default resource setting happens in both API and SDK, but the value from SDK takes more precedence.

You can refer here: https://github.com/caraml-dev/merlin/blob/main/api/service/version_endpoint_service.go#L115-L120 and here: https://github.com/caraml-dev/merlin/blob/main/api/service/version_endpoint_service.go#L130-L133

Where for every deployment of model version (the first one or redeployment), the resource request supplied by SDK (or UI) will be used to override the default environment's resource request value.

Reply:
I also realized that my first answer to your first question is wrong.

-> is it possible for user to not set min replica and set max replica?
Yes, it's possible. And it might not be using the default environment's default resource request value at all. Because the check in merlin-sdk is only check the whole ResourceRequest object not the max_replicas value: https://github.com/caraml-dev/merlin/blob/main/python/sdk/merlin/model.py#L1265

Reply:
I've updated the PR so that in sdk's deploy() function it will fallback to the default environment's resource request min and max replicas if these values are None

Reply:
can you refer me the one in sdk? i saw in [here](https://github.com/caraml-dev/merlin/blob/main/python/sdk/merlin/model.py#L1247) but the validation is conducted on different [object](https://github.com/caraml-dev/merlin/blob/main/python/sdk/merlin/model.py#L1266)

Reply:
nvm it is already addressed